app-desktop/build.gradle.kts

```diff
@@ -1,45 +1,129 @@
 plugins {
-    alias(libs.plugins.kotlinJvm)
+    alias(libs.plugins.kotlinMultiplatform)
     alias(libs.plugins.jetbrainsCompose)
     alias(libs.plugins.composeCompiler)
     alias(libs.plugins.composeHotReload)
     id("droidkaigi.primitive.metro")
     id("droidkaigi.primitive.aboutlibraries")
     id("droidkaigi.primitive.spotless")
+    id("droidkaigi.primitive.kmp.compose.resources")
+}
+
+compose {
+    resources {
+        nameOfResClass = "AppDesktopRes"
+    }
 }
 
 val aboutLibrariesTargetDir = "${layout.buildDirectory.get().asFile.path}/generated/aboutlibraries"
 
 kotlin {
+    jvm("desktop")
+
+    // JDK Version 21 is required to create the distribution materials.
+    jvmToolchain(21)
+
     compilerOptions {
         freeCompilerArgs.add("-Xcontext-parameters")
     }
 
     sourceSets {
-        main {
+        val commonMain by getting {
+            dependencies {
+                implementation(projects.appShared)
+                implementation(projects.core.data)
+                implementation(compose.runtime)
+                implementation(compose.foundation)
+                implementation(compose.ui)
+            }
+        }
+        val desktopMain by getting {
+            dependencies {
+                implementation(compose.desktop.currentOs)
+                implementation(libs.navigationeventDesktop)
+                implementation(libs.lifecycleRuntimeCompose)
+                implementation(compose.uiUtil)
+            }
             resources.srcDir(aboutLibrariesTargetDir)
         }
     }
 }
 
 compose.desktop.application.mainClass = "io.github.droidkaigi.confsched.MainKt"
 
-dependencies {
-    implementation(projects.appShared)
-    implementation(projects.core.data)
-    implementation(compose.desktop.currentOs)
-    implementation(compose.foundation)
-    implementation(compose.runtime)
-    implementation(libs.navigationeventDesktop)
-    implementation(libs.lifecycleRuntimeCompose)
-    implementation(compose.ui)
-    implementation(compose.uiUtil)
-}
-
 aboutLibraries.export {
     outputFile.set(file("${aboutLibrariesTargetDir}/licenses.json"))
 }
 
-tasks.named("processResources") {
-    dependsOn("exportLibraryDefinitions")
+compose.desktop {
+    application {
+        nativeDistributions {
+            targetFormats(
+                // for MacOS
+                org.jetbrains.compose.desktop.application.dsl.TargetFormat.Dmg,
+                // for Windows
+                org.jetbrains.compose.desktop.application.dsl.TargetFormat.Msi,
+                org.jetbrains.compose.desktop.application.dsl.TargetFormat.Exe,
```

Comment:
```shell
conference-app-2025/app-desktop/build/compose/binaries/main/dmg or app
```

The distribution files will be created in the above directory.
The directory is the same for both Mac and Windows.

Reply:
In my environment, both Windows and Mac are configured as follows. 💁‍♂️ 

```shell
openjdk 21.0.8 2025-07-15 LTS
OpenJDK Runtime Environment Temurin-21.0.8+9 (build 21.0.8+9-LTS)
OpenJDK 64-Bit Server VM Temurin-21.0.8+9 (build 21.0.8+9-LTS, mixed mode, sharing)
```

Indeed, I confirmed that even after removing the jvmToolchain specification in my own environment, the command succeeded on both Mac and Windows. 🙏 
I removed it in the following commit. 🫡 

https://github.com/DroidKaigi/conference-app-2025/pull/320/commits/e1503201ba5defc0516f56bb227aa85f4094e9e7